src/application/can_handler/can_handler.c

```diff
@@ -0,0 +1,98 @@
+#include "can_handler.h"
+
+QueueHandle_t busQueue_rx;
+QueueHandle_t busQueue_tx;
+
+can_callback_t callback_map[30];
+
+void can_handle_rx_isr(const can_msg_t *message, uint32_t timestamp) {
+    // The timestamp parameter passed to the handler is some internal FDCAN thing that I don't know
+    // how to convert to a sensible value Just use millis_() for now
+    xQueueSend(busQueue_rx, &message, 10);
+    return;
+}
+
+w_status_t can_register_callback(can_msg_type_t msg_type, can_callback_t callback) {
+    callback_map[msg_type] = callback;
+    return W_SUCCESS;
+}
+
+w_status_t can_handle_tx(const can_msg_t *message) {
+    if (xQueueSend(busQueue_tx, message, 10)) {
+        return W_SUCCESS;
+    } else {
+        return W_FAILURE;
+    }
+}
+
+void can_handler_task_rx(void *argument) {
+    busQueue_rx = xQueueCreate(16, sizeof(can_msg_t));
+    bool temp = can_init_stm(&hfdcan1, can_handle_rx_isr);
+    for (;;) {
+        can_msg_t *rx_msg;
+        if (xQueueReceive(busQueue_rx, &rx_msg, 10) == pdTRUE) {
+            uint16_t msgType = get_message_type(rx_msg);
+            if (callback_map[msgType] != NULL) {
+                callback_map[msgType](rx_msg);
+            }
+        }
+    }
+}
+
+void can_handler_task_tx(void *argument) {
+    busQueue_tx = xQueueCreate(16, sizeof(can_msg_t));
+    for (;;) {
+        can_msg_t tx_msg;
+        // Block the thread until we see data in the bus queue or 1 sec elapses
+        if (xQueueReceive(busQueue_tx, &tx_msg, 1000) ==
+            pdTRUE) { // Returns pdTRUE if we got a message, pdFALSE if timed out
+            if (can_send(&tx_msg) == false) {
+                // logError("CAN", "CAN send failed!");
+                int a = 1;
+            } else {
+                int b = 1;
+            }
+            vTaskDelay(1);
+        }
+    }
+}
+
+w_status_t can_handler_init(void) {
+    busQueue_rx = xQueueCreate(16, sizeof(can_msg_t));
+    busQueue_tx = xQueueCreate(16, sizeof(can_msg_t));
+
+    if (busQueue_tx == NULL) {
+        return W_FAILURE;
+    }
+    if (busQueue_rx == NULL) {
+        return W_FAILURE;
+    }
+    if (!can_init_stm(&hfdcan1, can_handle_rx_isr)) {
+        return W_FAILURE;
+    }
+    return W_SUCCESS;
```

Comment:
We should also bind the callback for reset commands `MSG_RESET_CMD` here, since there isn't a better place to put it and we want to be able to reset so long as CAN is working